internal/provider/provider_test.go

```diff
@@ -88,7 +92,11 @@ func testMethodResource(method string, urlPath string) ([]byte, error) {
 	var body []byte
 	switch method {
 	case http.MethodGet:
-		body, diags = client.Get(urlPath)
+		if params != nil {
+			body, diags = client.GetWithParams(urlPath, params)
+		} else {
+			body, diags = client.Get(urlPath)
+		}
```

Comment:
I see you abstracted `testMethodResource` to `testMethodResourceWithParams` so that we could supply params to a GET request. I didn't find any changes in this PR that use the new `testMethodResourceWithParams`, just `testMethodResource` (and that passes `nil` as the params). 

So why the change? Was this leftover from a test that required `testMethodResource` with params?